### DISCUSSION TITLE
Vercel Edge Cache based on `x-vercel-ip-country-region` request header value.

### DISCUSSION THREAD

**Discussion by prashantchothani** [vercel/vercel#10906] [Ideas]
### Description

We are getting the header `x-vercel-ip-country-region` for every request. We want that responses should be cached by country.

How do achieve this in Vercel using Nextjs where `x-vercel-ip-country-region` request header if it contains IN, the cached response should return where the request was having IN as the header. Similarly if the user accesses this service from US, the  `x-vercel-ip-country-region` request header value will be US and it should have a different cache if it exists. If not, the API would fetch the data and cache the same. This cached response should be unique for the requests that have the `x-vercel-ip-country-region` value as US.  In short, we want the Cache to be unique to each `x-vercel-ip-country-region` value of the request header.

For reference, on AWS Cloudfront there is a similar functionality like above where we can cache based on the request header value. The below is the link for the same:
https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/header-caching.html#header-caching-web-location

**Comment by leerob** [comment]
Hey there. It is not possible to cache based on the country currently on Vercel. We would like to support the `Vary` header for caching responses: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Vary

**Reply by prashantchothani** [reply]
@leerob Is this on Vercel Roadmap anytime ?

**Reply by prashantchothani** [reply]
How can we achieve something like this using a vary header ? And will it work on Vercel? The below link shows a section of having cache by country using vary cache. This will solve our problem and be a huge relief @leerob if you can suggest a solution with Vercel.

https://support.bunny.net/hc/en-us/articles/360020604140-Understanding-Vary-Cache
